cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SecurityUserAndRoleManagementTest.scala

```diff
@@ -147,9 +147,11 @@ class SecurityUserAndRoleManagementTest extends DocumentingTest with QueryStatis
     }
     section("Role Management", "administration-security-roles", "enterprise-edition") {
       initQueries("CREATE USER jake SET PASSWORD 'abc123' CHANGE NOT REQUIRED",
-        "CREATE USER user1 SET PASSWORD 'abc'", "CREATE USER user2 SET PASSWORD 'abc'", "CREATE USER user3 SET PASSWORD 'abc'",
-        "CREATE ROLE role1", "CREATE ROLE role2")
+        "CREATE USER user1 SET PASSWORD 'abc'", "CREATE USER user2 SET PASSWORD 'abc'", "CREATE USER user3 SET PASSWORD 'abc'")
       p("Roles can be created and managed using a set of Cypher administration commands executed against the `system` database.")
+      important {
+        p("""It is not allowed to use the role name `PUBLIC`. This is a system reserved role that should not be used.""")
```

Comment:
Is this advisory? as-in, users can use the role name `PUBLIC`, but we're asking them not to, OR users simply cannot use the role name `PUBLIC` as it's reserved for the system database?

I think if it's the 2nd instance, we can reword to:

> The role name `PUBLIC` is not permitted. This is a system-reserved role, and cannot be used.

Reply:
In 4.0.0 they can use `PUBLIC` because we forgot to reserve it. In 4.0.1 we have reserved it so that users will not be able to create it AND if they are upgrading from 4.0.0 while having it, the upgrade will fail and Neo4j will not start.

I think your suggestion fits nicely